### PR TITLE
Adding ServiceBusManagementClient and builder.

### PR DESCRIPTION
**Pull request by conniey**
Adding ServiceBusManagementClient, ServiceBusManagementAsyncClient, and ServiceBusManagementClientBuilder.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+                    return new SimpleResponse<>(response.getRequest(), response.getStatusCode(),
+                        response.getHeaders(), null);
+                });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Package-private method that gets a queue with its context.
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementAsyncClient.java:294]: `Package-private method ` : I have never seen that writing in java doc, it is evident from the fact that there is no qualifier. 
I am fine with this but it is not usual.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClient.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+     * @param queue Information about the queue to create.
+     *
+     * @return The created queue.
+     * @throws NullPointerException if {@code queue} is null.
+     * @throws IllegalArgumentException if {@link QueueDescription#getName() queue.getName()} is null or an empty
+     *     string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public QueueDescription createQueue(QueueDescription queue) {
+        return asyncClient.createQueue(queue).block();
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClient.java:48]: `queueDescription`  as  arg name will be more appropriate ?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClientBuilder.java [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+    /**
+     * Sets the service endpoint for the Service Bus namespace.
+     *
+     * @param endpoint The URL of the Service Bus namespace.
+     *
+     * @return The updated {@link ServiceBusManagementClientBuilder} object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public ServiceBusManagementClientBuilder endpoint(String endpoint) {
+        try {
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClientBuilder.java:144]: Should we check for `endpoint` could be null and throw NPE ?

Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClientBuilder.java:144]: It checks on buildClient

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClient.java (hunk 2 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusManagementClient.java:48]: Went with .NET's parameter name.

**Pull request merged**